go/chat/boxer.go

```diff
@@ -1325,11 +1465,35 @@ func (b *Boxer) boxV1(messagePlaintext chat1.MessagePlaintext, key types.CryptKe
 	return boxed, nil
 }
 
-// V3 is just V2 but with exploding messages support. All future versions after
-// V3 will support exploding messages without a version distinction.
+func makeOnePairwiseMAC(private libkb.NaclDHKeyPrivate, public libkb.NaclDHKeyPublic, input []byte) ([]byte, error) {
+	// Compute the shared DH secret, and then mix some context into it. In
+	// other places where we derive keys, we've used HMAC with the context
+	// string as the message. In this case though, because the raw shared
+	// secret is so general, that seems *particularly* likely to collide with
+	// some unexpected use of HMAC elsewhere. To avoid that possibility, we
+	// stick to a simple concatenation-plus-hashing derivation. (Bleeding edge
+	// hash functions like BLAKE2 accept a context string directly, which is a
+	// much better solution. We can dream.)
+	privKeyBytes := [32]byte(private)
+	pubKeyBytes := [32]byte(public)
+	var rawShared [32]byte
+	box.Precompute(&rawShared, &pubKeyBytes, &privKeyBytes)
+	deriveState := sha256.New()
+	deriveState.Write([]byte(libkb.DeriveReasonChatPairwiseMAC))
+	deriveState.Write([]byte{0})
+	deriveState.Write(rawShared[:])
+	derivedShared := deriveState.Sum(nil)
+
+	hmacState := hmac.New(sha256.New, derivedShared)
+	hmacState.Write(input)
+	return hmacState.Sum(nil), nil
```

Comment:
can we just return `[]btye`?

Reply:
Woops yes, an earlier version of this function included some key parsing code.